opamp/observiq/agent_config_manager.go

```diff
@@ -0,0 +1,167 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observiq
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/observiq/observiq-otel-collector/opamp"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"go.uber.org/zap"
+)
+
+const (
+	// CollectorConfigName is the key of the collector config in OpAmp
+	CollectorConfigName = "collector.yaml"
+	// ManagerConfigName is the key of the manager config in OpAmp
+	ManagerConfigName = "manager.yaml"
+	// LoggingConfigName is the key of the logging config in OpAmp
+	LoggingConfigName = "logging.yaml"
+)
+
+// acceptableConfigs is a lookup of configs that are able to be written/updated
+var acceptableConfigs = map[string]struct{}{
```

Comment:
won't hold up the PR if the answer is yes (as long as we make a tech debt bug), but should acceptableConfigs be used in the other two public methods as well?

Reply:
`acceptableConfigs` is used as a quick reference/security check to verify the config the server is sending down to update/add is one of the supported ones. We only use this for that check. It doesn't matter to use if you add more than these to be managed but they won't be able to be remote managed unless added to this list.

Reply:
@cpheps practically though, will we even be able to run with unacceptable configs right now (if running in managed mode)?

or another question. what happens if we have an unacceptable config which gets sent back to iris. what does that mean?

Reply:
Talked this through on slack. I don't think this is a current concern but could be in the future.